spec/cli_spec.rb

```diff
@@ -528,6 +528,81 @@
     end
   end
 
+  context "#set_replication?" do
```

Comment:
I was thinking something more targeted for these.

For example:

```ruby
  it "returns false when cluster node number is not set" do
    subject.options = {:replication => "primary", :password => "password"}
    expect(subject.set_replication?).to be_falsey
  end

  it "returns false when the replication option is set to an invalid value" do
    #more test stuff here
  end
```

This way you don't have to deal with the exception and you are not also testing `#parse` and `#run`.